### PR TITLE
Remove FILTERED

### LINKED ISSUE #76: Change returned filter panel code to avoid `_FILTERED` suffix
$get_call should return following code 

```r
ADSL <- filter(ADSL, <filters>)
ADTTE <- filter(ADTTE, <filters>)
ADTTE <- inner_join(ADTTE, ADSL, by = <keys>)
...
```

This will require also to make a change in https://github.com/insightsengineering/teal.transform/search?q=_FILTERED

Please add NEWS.md entry under BREAKING CHANGE.

### PR DESCRIPTION
**Pull request by nikolas-burkoff**
Closes #76 

Also have:
- https://github.com/insightsengineering/teal.data/pull/90
- https://github.com/insightsengineering/teal.transform/pull/87 
- https://github.com/insightsengineering/teal/pull/702
- https://github.com/insightsengineering/teal.modules.clinical/pull/536
- https://github.com/insightsengineering/teal.goshawk/pull/174
- https://github.com/insightsengineering/teal.osprey/pull/166
- https://github.com/insightsengineering/teal.modules.general/pull/431
- tmh looks good

Should we have a warning in teal.code?

Test with examples in vignettes and also with these apps

```
library(teal)
library(scda)

ADSL <- synthetic_cdisc_data("latest")$adsl
ADAE <- synthetic_cdisc_data("latest")$adae

app <- init(
  data = cdisc_data(
    cdisc_dataset("ADSL", ADSL, code = "ADSL <- synthetic_cdisc_data(\"latest\")$adsl"),
    cdisc_dataset("ADAE", ADAE, code = "ADAE <- synthetic_cdisc_data(\"latest\")$adae")
  ),
  modules = modules(teal:::filter_calls_module())
)

shinyApp(app$ui, app$server)


utils::data(miniACC, package = "MultiAssayExperiment")

app <- init(
  data = teal_data(
    dataset("iris", iris),
    dataset("mtcars", mtcars),
    dataset("miniacc", miniACC)
  ),
  modules = modules(teal:::filter_calls_module())
)

shinyApp(app$ui, app$server)

```

### CODE REVIEW

**Review by gogonzo — COMMENTED** [review]
I don't see anything here to fix except these minor comments.

File: R/CDISCFilteredData.R [hunk 1]
```diff
@@ -97,9 +97,8 @@ CDISCFilteredData <- R6::R6Class( # nolint
 
         dataset <- self$get_filtered_dataset(dataname)
 
-        filtered_dataname <- dataset$get_filtered_dataname()
-        filtered_dataname_alone <- dataset$get_filtered_dataname(suffix = "_FILTERED_ALONE")
-        filtered_parentname <- dataset$get_filtered_dataname(dataname = parent_dataname)
+        filtered_dataname <- dataname
+        filtered_parentname <- parent_dataname
```
[the remark this answers is not in the document]
Comment (nikolas-burkoff) [reply R/CDISCFilteredData.R:101]: done

File: R/CDISCFilteredData.R [hunk 2]
```diff
@@ -108,13 +107,13 @@ CDISCFilteredData <- R6::R6Class( # nolint
...
         )
-        premerge_call[[1]][[2]] <- as.name(filtered_dataname_alone)
+
         merge_call <- call(
           "<-",
           as.name(filtered_dataname),
           call_with_colon(
             "dplyr::inner_join",
-            x = as.name(filtered_dataname_alone),
+            x = as.name(filtered_dataname),
```
[the remark this answers is not in the document]
Comment (nikolas-burkoff) [reply R/CDISCFilteredData.R:116]: done

**Review by gogonzo — APPROVED** [review]
Everything is good here.
No `_FILTERED` in whole insightsengineering. Apps working correctly

**Pull request merged**